exporter/collector/monitoredresource_test.go

```diff
@@ -0,0 +1,269 @@
+				"cloud.platform":          "fooprovider_kubernetes_service",
+				"cloud.availability_zone": "us-central1",
+				"service.namespace":       "myservicenamespace",
+				"service.name":            "myservicename",
+				"service.instance.id":     "myserviceinstanceid",
+			},
+			expectMr: &monitoredrespb.MonitoredResource{
+				Type: "generic_task",
+				Labels: map[string]string{
+					"job":       "myservicename",
+					"location":  "us-central1",
+					"namespace": "myservicenamespace",
+					"task_id":   "myserviceinstanceid",
+				},
+			},
+			expectExtraLabels: labels{},
+		},
+		{
+			name: "Generic task without cloud.availability_zone region",
+			resourceLabels: map[string]string{
+				"cloud.platform":      "fooprovider_kubernetes_service",
+				"cloud.region":        "my-region",
+				"service.namespace":   "myservicenamespace",
+				"service.name":        "myservicename",
+				"service.instance.id": "myserviceinstanceid",
+			},
+			expectMr: &monitoredrespb.MonitoredResource{
+				Type: "generic_task",
+				Labels: map[string]string{
+					"job":       "myservicename",
+					"location":  "my-region",
+					"namespace": "myservicenamespace",
+					"task_id":   "myserviceinstanceid",
+				},
+			},
+			expectExtraLabels: labels{},
+		},
+		{
+			name: "Generic node",
+			resourceLabels: map[string]string{
+				"cloud.platform":          "fooprovider_kubernetes_service",
+				"cloud.availability_zone": "us-central1",
+				"cloud.region":            "my-region",
+				"service.namespace":       "myservicenamespace",
+				"service.name":            "myservicename",
+				"host.id":                 "myhostid",
+				"host.name":               "myhostname",
+			},
+			expectMr: &monitoredrespb.MonitoredResource{
+				Type: "generic_node",
+				Labels: map[string]string{
+					"location":  "us-central1",
+					"namespace": "myservicenamespace",
+					"node_id":   "myhostid",
+				},
+			},
+			expectExtraLabels: labels{},
+		},
+		{
+			name: "Generic node without cloud.availability_zone",
+			resourceLabels: map[string]string{
+				"cloud.region":      "my-region",
+				"service.namespace": "myservicenamespace",
+				"host.id":           "myhostid",
+				"host.name":         "myhostname",
+			},
+			expectMr: &monitoredrespb.MonitoredResource{
+				Type: "generic_node",
+				Labels: map[string]string{
+					"location":  "my-region",
+					"namespace": "myservicenamespace",
+					"node_id":   "myhostid",
+				},
+			},
+			expectExtraLabels: labels{},
+		},
+		{
+			name: "Generic node without host.id",
+			resourceLabels: map[string]string{
+				"cloud.region":      "my-region",
+				"service.namespace": "myservicenamespace",
+				"host.name":         "myhostname",
+			},
+			expectMr: &monitoredrespb.MonitoredResource{
+				Type: "generic_node",
+				Labels: map[string]string{
+					"location":  "my-region",
+					"namespace": "myservicenamespace",
+					"node_id":   "myhostname",
+				},
+			},
+			expectExtraLabels: labels{},
+		},
```

Comment:
It would be good to add a test where the fall through to generic_node populates the labels with empty string.

Reply:
Done